### DISCUSSION TITLE
pnpm usage with depcheck?

### DISCUSSION THREAD

**Discussion by jasperdunn** [pnpm/pnpm#4568] [Q&A] [answered]
As I understand it, `pnpm` by default requires that all `peerDependencies` of our `package.json:"dependencies"` be installed.
The default configuration also doesn't install these automatically, which means that during build, things break unless you resolve these manually. Not sure if doing something wrong here?

[depcheck](https://github.com/depcheck/depcheck#readme) retrieves a list of packages that aren't currently being used in your project. When run on a project using `pnpm`, it returns these `peerDependencies` packages as "Unused dependencies".

Just wondering if there is a recommended approach to this.
Thanks

**Comment by zkochan** [comment, marked as answer]
I don't think depcheck should report such dependencies.

I see a related issue in depcheck, maybe it'll help: https://github.com/depcheck/depcheck/issues/663

**Reply by jasperdunn** [reply]
Yikes that thread got shouty! Thanks, yeah I guess we'll have to wait and see.

**Reply by YenHub** [reply]
Crikey, that aged well! 

<img width="1045" height="281" alt="image" src="https://github.com/user-attachments/assets/552e0df7-5989-4b3c-a4ba-ce82a93990ee" />
